### ISSUE TITLE
App doesn't stay activated after disconnection from PC

### ISSUE THREAD

**Issue by Spir8** [dysquard/pgpa#44]
Hello,
I've got a Huawei Mate 20 X with Android 10. The activation process works, but as soon as I unplug my phone from my PC the app loses its activation. I've tried both Activate.bat and ActivateWifi.bat files, ran them as Administrator, ran them straight after eachother, rebooted my phone, reinstalled the app, disabled battery savings options on my phone for the Panda app, ...
Please help.

**Comment by Spir8** [comment]
Well I've seen to have fixed that issue. Next problem is that the keyboard doesn't work. I've tried different keyboards, bluetooth and wired, ... When in Android they all work, but in a game with Panda running they don't detect any input whatsoever.
I've tried the Octopus app, it works perfectly in that app.

**Comment by mitolohiya** [comment]
> Well I've seen to have fixed that issue. Next problem is that the keyboard doesn't work. I've tried different keyboards, bluetooth and wired, ... When in Android they all work, but in a game with Panda running they don't detect any input whatsoever.
> I've tried the Octopus app, it works perfectly in 

> Hello,
> I've got a Huawei Mate 20 X with Android 10. The activation process works, but as soon as I unplug my phone from my PC the app loses its activation. I've tried both Activate.bat and ActivateWifi.bat files, ran them as Administrator, ran them straight after eachother, rebooted my phone, reinstalled the app, disabled battery savings options on my phone for the Panda app, ...
> Please help.

How you fix the problem about not staying activated for a few minutes???

**Comment by Spir8** [comment]
> How you fix the problem about not staying activated for a few minutes???
You have to activate it when your phone is in charging only mode, not in file transfer mode (when you connect your phone to your pc with USB). And then you also have to enable the option in Developer Mode to let USB debugging happen when in Charge Only mode.
That did it for me.

**Comment by mitolohiya** [comment]
> You have to activate it when your phone is in charging only mode, not in file transfer mode (when you connect your phone to your pc with USB). And then you also have to enable the option in Developer Mode to let USB debugging happen when in Charge Only mode.
> That did it for me.

But when I connect it with only charge only mode, my phone can't detect

**Comment by Spir8** [comment]
Good that's what you want. So you can't transfer files from your pc to your phone through USB?

**Comment by mitolohiya** [comment]
> Good that's what you want. So you can't transfer files from your pc to your phone through USB?

i want to activate the panda mouse pro, but when I activate it with transfer file successful in a few minutes it will need to activate again and I can't play but you said you fix the issue by activating with charge only, but if I put in charge only, my phone not found just like on the photo
![15935254443915493549766925395900](https://user-images.githubusercontent.com/67642029/86134881-b2b66e00-bb1c-11ea-9d4f-341032e6abc5.jpg)

**Comment by Spir8** [comment]
You have to go to developer settings and find an option that allows USB debugging in charge only mode. Activate that, then make sure USB debugging is on, and it should work.

**Comment by mitolohiya** [comment]
> You have to go to developer settings and find an option that allows USB debugging in charge only mode. Activate that, then make sure USB debugging is on, and it should work.

Sorry, but I don't get it. There is only one usb debugging option where you on/off, can you send a photo? I really want to use the panda mouse pro bc I recently bought a usb hub.

**Comment by Spir8** [comment]
![2020-07-01 at 14 36 59](https://user-images.githubusercontent.com/65722746/86250447-7e888f00-bba8-11ea-8766-78e0780d4b64.jpeg)

**Comment by mitolohiya** [comment]
![Screenshot_2020-07-01-21-40-42-53](https://user-images.githubusercontent.com/67642029/86250873-ba8d1580-bbe3-11ea-986b-ce18471a141a.jpg)
 i
![Screenshot_2020-07-01-21-40-55-11](https://user-images.githubusercontent.com/67642029/86250881-bd880600-bbe3-11ea-99ec-b4436dd0a484.jpg) 

I don't have that

**Comment by Spir8** [comment]
Weird. Maybe some phones just don't allow USB debugging over charge only mode.
What happens when you enable USB debugging, go out of Developer Options, then connect your phone to your PC, select charge only mode, and go back to Developer Options. Is USB Debugging still enabled?
If not, enable it again, go out of Developer Options, and go back in. What's the status now of the USB Debugging?

**Comment by mitolohiya** [comment]
> Weird. Maybe some phones just don't allow USB debugging over charge only mode.
> What happens when you enable USB debugging, go out of Developer Options, then connect your phone to your PC, select charge only mode, and go back to Developer Options. Is USB Debugging still enabled?
> If not, enable it again, go out of Developer Options, and go back in. What's the status now of the USB Debugging?

It still enabled, but earlier in the morning, I activated the panda mouse using the activatewifi luckily but the keyboard is not working only mouse, when playing pubg through panda mouse, and when I turned off the usb debug the panda mouse deactivated. I think it because of Android 10 and now I'm resetting my phone so that it can turn back into android 9 and I will not update it.

**Comment by Spir8** [comment]
Yeah as I've said in my original post, I've got the same issue with Android 10 and the keyboard. What phone are you using? I'm on a Huawei Mate 20X.
I'm just hoping they're gonna release an update that fixes this issue because I prefer to stay on Android 10.

**Comment by mitolohiya** [comment]
> Yeah as I've said in my original post, I've got the same issue with Android 10 and the keyboard. What phone are you using? I'm on a Huawei Mate 20X.
> I'm just hoping they're gonna release an update that fixes this issue because I prefer to stay on Android 10.

I'm using realme 5 pro, yeah I prefer Android 10 too, btw thank you for helping me
